### ISSUE TITLE
Audio Recording stops working when External MQTT is selected 

### ISSUE THREAD

**Issue by Rum459** [rhasspy/rhasspy#10]
Settings:
  MQTT: **Internal**
  Intent Handling: **Disabled**
  Wake Word: **Snowboy**
  Record Audio: **PyAudio** with Seeed-2mic-voicecard: - (hw:0,0) 
  Speech To Text: **PocketSphinx**
  Intent Recognition: **Fsticuffs**
  Text To Speech: **PicoTTS**
  Audio Playing: **aplay**
  Dialogue Management: **Rhasspy**

When MQTT is configured for External the wake word stops functioning and the Record Device list only shows Default Device (instead of Seeed-2mic-voicecard: - (hw:0,0)  ).

**Comment by joergeli** [comment]
Seems that I have the same problem.
Take a look at:
https://community.rhasspy.org/t/rhasspy-2-5-pre-release/643/2

**Comment by joergeli** [comment]
No way to fix the external MQTT-broker-problem?
Under these circumstances I unfortunately have no opportunity to test rhasspy 2.5-pre

**Comment by joergeli** [comment]
Perhaps a log could help with the external MQTT-broker problem?
Here is a log, where I've just pressed the wake-button ( no beeps to hear).
For me, it seems that  AsrStartListening starts but because micro is "dead" no command is recognized and after a timeout of ~30 seconds rhasspyserver_hermes throws an error:
```
[ERROR:2020-04-07 19:01:56,380] rhasspyserver_hermes: 
Traceback (most recent call last):
  File "/usr/lib/rhasspy-voltron/.venv/lib/python3.7/site-packages/quart/app.py", line 1821, in full_dispatch_request
    result = await self.dispatch_request(request_context)
  File "/usr/lib/rhasspy-voltron/.venv/lib/python3.7/site-packages/quart/app.py", line 1869, in dispatch_request
    return await handler(**request_.view_args)
  File "/usr/lib/rhasspy-voltron/rhasspy-server-hermes/rhasspyserver_hermes/__main__.py", line 811, in api_listen_for_command
    handle_captured(), messages, message_types
  File "/usr/lib/rhasspy-voltron/rhasspy-server-hermes/rhasspyserver_hermes/__init__.py", line 854, in publish_wait
    result_awaitable, timeout=timeout_seconds
  File "/usr/lib/python3.7/asyncio/tasks.py", line 449, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError
[ERROR:2020-04-07 19:01:26,383] rhasspyserver_hermes: on_message
Traceback (most recent call last):
  File "/usr/lib/rhasspy-voltron/rhasspy-server-hermes/rhasspyserver_hermes/__init__.py", line 959, in on_message
    topic, payload, self.subscribed_types
  File "/usr/lib/rhasspy-voltron/rhasspy-hermes/rhasspyhermes/client.py", line 269, in parse_mqtt_message
    json_payload = json.loads(payload)
  File "/usr/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
[DEBUG:2020-04-07 19:01:26,379] rhasspyserver_hermes: Publishing 142 bytes(s) to hermes/asr/startListening
[DEBUG:2020-04-07 19:01:26,378] rhasspyserver_hermes: -> AsrStartListening(siteId='default', sessionId='8337c6c7-38cb-4bee-b7e9-f46a9a6e38f7', stopOnSilence=True, sendAudioCaptured=True, wakewordId='')
[DEBUG:2020-04-07 19:01:26,377] rhasspyserver_hermes: Subscribed to hermes/error/asr
[DEBUG:2020-04-07 19:01:26,377] rhasspyserver_hermes: Waiting for transcription (sessionId=8337c6c7-38cb-4bee-b7e9-f46a9a6e38f7)
```
And a screenshot with my configuration:
Rhasspy V2.5-pre runs on a Pi4 in docker-container
![config](https://user-images.githubusercontent.com/62521927/78709523-15301180-7914-11ea-978e-5ed8635e58f6.jpg)

Greetings
Jörg

**Comment by synesthesiam** [comment]
Your error suggests a malformed JSON message. Let me add some more logging the rhasspy-server so we can figure out what that message is.

**Comment by joergeli** [comment]
Great!
Perhaps with a detailed logging we can fix the problem

Thanks in advance
Jörg

**Comment by capitaenz** [comment]
@synesthesiam 
Is there a solution to this problem or a workaround?

**Comment by synesthesiam** [comment]
Sorry for the late reply. Is this still happening with the latest release?

**Comment by Rum459** [comment]
No, it was resolved shortly after reporting the issue. The current release works great! 

Thanks

**Comment by joergeli** [comment]
@ synesthesiam
I had to do a workaround, to get latest Rhasspy 2.5 (stable) running:
Installing a standalone MOSQUITTO-Broker on the the same Raspi, where Rhasspy is running in a docker-container.
Set MQTT-settings in Rhasspy: MQTT = external, IP= IP of Rhasspy, Port 1883.
Then subscribing with my MQTT-Client to the MOSQUITTO.
After that Wakeword is recognized and Intent Recognition works.

So I dont't really know, If I understod something wrong with my further MQTT-configuration.
In other words: It's not possible anymore to subscribe directly to the **_internal_** Rhasspy MQTT-Broker, because it is running in a docker-container (in contrast to Rhasspy V2.4.19) ?

Anyway, with the additional MOSQUITTO it works.

Thanks
Jörg

**Comment by synesthesiam** [comment]
The internal broker runs on port 12183, so if you're running in a Docker container just make sure to add `-p 12183:12183` to your `docker run` command. Of course, you'll need to adjust your MQTT client ports too :)

**Comment by Romkabouter** [comment]
@synesthesiam can be closed 👍
